File: 300/d/main.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

const long long Z = 1000000;
bool isprime[Z+1] = {};
int count_primes[Z+1] = {};
vector<long long> primes;


int main(){
    for(int p=2; p<=Z; p++) isprime[p] = true;
    for(int p=2; p*p<=Z; p++) if(isprime[p]){
        for(int q=p*p; q<=Z; q+=p) isprime[q] = false;
    }
    for(int p=2; p<=Z; p++) if(isprime[p]){
        primes.push_back(p);
        count_primes[p] = 1;
    }
    for(int p=2; p<=Z; p++) count_primes[p] += count_primes[p-1];

    long long N; cin >> N;
    long long ans = 0;
    for(long long a=2; a*a*a*a*a<=N; a++) if(isprime[a]) {
        for(long long c=a+1; a*a*a*c*c<=N; c++) if(isprime[c]) {
            long long br = min(c-1, N/(a*a*c*c));
            long long bl = a;
            if(bl < br) {
                ans += count_primes[br];
                ans -= count_primes[bl];
            }
        }
    }
    cout << ans << endl;
    return 0;
}

File: 300/d/main.py
N = int(input())


def sieve(n):
    is_prime = [True] * (n + 1)
    is_prime[0] = False
    is_prime[1] = False
    primes = []
    for i in range(2, n + 1):
        if is_prime[i]:
            primes.append(i)
            for j in range(i * i, n + 1, i):
                is_prime[j] = False
    return primes

primes = sieve(int(N ** 0.5) + 1)  # Nの平方根までの素数を列挙
ans = 0
for i in range(len(primes)):
    for j in range(i + 1, len(primes)):
        a = primes[i]
        b = primes[j]
        if N // (a ** 2 * b) < b:
            break
        for k in range(j + 1, len(primes)):
            c = primes[k]
            if N < a ** 2 * b * c ** 2:
                break
            ans += 1
print(ans)
